src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -172,4 +176,16 @@ public String getOptionalArrayTypeName() {
   public String getDynamicReturnTypeName(Method method, MethodConfig methodConfig) {
     return getNotImplementedString("SurfaceNamer.getDynamicReturnTypeName");
   }
+
+  public String getApiSettingsClassName(Interface service) {
+    return className(Name.upperCamel(service.getSimpleName(), "Settings"));
+  }
+
+  public String getTestClassName(Interface service) {
+    return className(Name.upperCamel(service.getSimpleName(), "Test"));
+  }
+
+  public String getMockServiceClassName(Interface service) {
+    return className(Name.upperCamel("Mock", service.getSimpleName()));
+  }
```

Comment:
In contrast with documenting the viewmodel classes (I'm dubious of the value of that), I can see more value in documenting these methods, particularly because they are a more likely point of exploration by a new implementer, and given that many concepts are in one place. I'll add comments in my Java MVVM PR ( https://github.com/googleapis/toolkit/pull/337 ).